core/trino-main/src/main/java/io/trino/operator/join/JoinHashSupplier.java

```diff
@@ -52,6 +52,31 @@
     private final Optional<JoinFilterFunctionFactory> filterFunctionFactory;
     private final List<JoinFilterFunctionFactory> searchFunctionFactories;
 
+    public static long getEstimatedAdditionalSizeInBytes(
+            int positionCount,
+            LongArrayList addresses,
+            List<List<Block>> channels,
+            Optional<Integer> sortChannel,
+            OptionalInt singleBigintJoinChannel,
+            HashArraySizeSupplier hashArraySizeSupplier)
+    {
+        long result = 0;
+        if (sortChannel.isPresent()) {
+            result += SortedPositionLinks.getEstimatedRetainedSizeInBytes(positionCount);
```

Comment:
so positions links are estimated *before* lookup source is build. When they are accounted for *after* lookup source is built (I don't think it's accounted for at all)?